extensions/amp-carousel/0.1/slidescroll.js

```diff
@@ -686,7 +686,10 @@ export class AmpSlideScroll extends BaseSlides {
    */
   showSlide_(newIndex) {
     const {noOfSlides_} = this;
-    newIndex = dev().assertNumber(newIndex);
+    if (!isFiniteNumber(newIndex)) {
+      dev().error('Attempted to show slide that is not a number: %s', newIndex);
+      return;
+    }
     if (
       newIndex < 0 ||
       newIndex >= noOfSlides_ ||
```

Comment:
I see this part checks that the argument fits into reasonable bounds and silently fails. I think this already catches cases for `Infinity`, `-Infinity`, `1 / 0`, etc. In that case the above can use `isNaN` instead of `isFiniteNumber` to cover the current errors we're seeing. WDYT?

Reply:
Fair enough, I used isFiniteNumber for parity with the other checks in the file.